tests/lib/rules/newline-before-return.js

```diff
@@ -112,77 +112,90 @@ ruleTester.run("newline-before-return", rule, {
+            errors: ["Expected newline before return statement."],
+            output: "function a() {\nfor (b in c) {\nd();\n\nreturn;\n}\n}"
         },
         {
             code: "function a() {\nfor (b of c) {\nd();\nreturn;\n}\n}",
             parserOptions: { ecmaVersion: 6 },
-            errors: ["Expected newline before return statement."]
+            errors: ["Expected newline before return statement."],
+            output: "function a() {\nfor (b of c) {\nd();\n\nreturn;\n}\n}"
         },
         {
             code: "function a() {\nif (b) {\nc();\n}\n//comment\nreturn b;\n}",
-            errors: ["Expected newline before return statement."]
-        },
-        {
-            code: "function a() {\nif (b) {\nc();\n}\n//comment\nreturn b;\n}",
-            errors: ["Expected newline before return statement."]
+            errors: ["Expected newline before return statement."],
+            output: "function a() {\nif (b) {\nc();\n}\n//comment\nreturn b;\n}"
         },
         {
             code: "function a() {\n/*comment\ncomment*/\nif (b) {\nc();\nreturn b;\n} else {\n//comment\n\nreturn d;\n}\n/*multi-line\ncomment*/\nreturn e;\n}",
-            errors: ["Expected newline before return statement.", "Expected newline before return statement."]
+            errors: ["Expected newline before return statement.", "Expected newline before return statement."],
+            output: "function a() {\n/*comment\ncomment*/\nif (b) {\nc();\n\nreturn b;\n} else {\n//comment\n\nreturn d;\n}\n/*multi-line\ncomment*/\nreturn e;\n}"
         },
         {
             code: "function a() {\nif (b) { return; } //comment\nreturn c;\n}",
-            errors: ["Expected newline before return statement."]
+            errors: ["Expected newline before return statement."],
+            output: "function a() {\nif (b) { return; } //comment\n\nreturn c;\n}"
         },
         {
             code: "function a() {\nif (b) { return; } /*multi-line\ncomment*/\nreturn c;\n}",
-            errors: ["Expected newline before return statement."]
+            errors: ["Expected newline before return statement."],
+            output: "function a() {\nif (b) { return; } /*multi-line\ncomment*/\nreturn c;\n}"
         },
         {
             code: "function a() {\nif (b) { return; }\n/*multi-line\ncomment*/ return c;\n}",
-            errors: ["Expected newline before return statement."]
+            errors: ["Expected newline before return statement."],
+            output: "function a() {\nif (b) { return; }\n/*multi-line\ncomment*/ return c;\n}"
         },
         {
             code: "function a() {\nif (b) { return; } /*multi-line\ncomment*/ return c;\n}",
-            errors: ["Expected newline before return statement."]
+            errors: ["Expected newline before return statement."],
+            output: "function a() {\nif (b) { return; } /*multi-line\ncomment*/ \nreturn c;\n}"
```

Comment:
This is the case where I was most in doubt. Here's a rundown:

Before fix, `return c;` is flagged as error:
```javascript
function a() {
    if (b) { return; } /*multi-line
    comment*/ return c;
}
```

After fix, `return c;` is still flagged as error, but no subsequent fix is applied:
```javascript
function a() {
    if (b) { return; } /*multi-line
    comment*/ 
    return c;
}
```

First, the rule logic marks that the multi-line comment belongs to the previous statement, meaning that the return statement is considered safe to fix.
After the newline is inserted, the rule will flag the same return statement again, but now it calculates that the return statement contains comments, hence a second fix will not be applied.
Any thoughts on this behavior?